# Future Directions

## 1. Slightly change the genome filtering methodology, to use the metric 'Complete BUSCO (%)' versus 'Single-copy BUSCO (%)', which is a more standard metric, and would allow for the addition of ~20 more genomes. The cutoff will still be 95%.

*Why do this?* 

1. In corporate more Peltigerales genomes. The *Lobraria* genomes (Peltigerales) which seem very different compared to all of the other lineages in the Lecanoromycetes.

2. Incorporate more genomes from the Pertusariaceae. It is currently just *Agyrium*, which makes comparsions difficult because it is unclear if the differences in orthogroup copy number are due to *Agyrium* being non-lichenized, or if this is a lineage difference.

**NEED TO CHECK IF OTHER LINEAGES ARE ALSO VERY UNDER-REPRESENTED AND SEE IF THERE IS A WAY TO INCORPORATE THESE**

## 2. Incorporate transcriptome (RNA-seq) information into the gene prediction and modeling

This has been shown to substantially impact the gene predictions and improves the capability to identify locations of introns and alternate splicing sites.

## 3. Incorpoorate genomes from the Eurotiomycetes and Lichinomycetes.

The Lecanoromyctes and Eurotiomycetes are sister clades. The L+E clade is sister to the Lichinomycetes. It is highly unclear the origins of lichenization within this triad. Notably, the Lichinomycetes represents a class of fungi that are almost entirelt symbiotic, as lichens, endophytes, and mycorrhizae.
