# Methodology

All publically available genomes representing Lecanoromcyetes fungi were catalogued from NCBI (accessed 02/05/2025) and MycoCosm (JGI; last accessed 04/28/2025). These genomes were catalouged with taxonomic information using NCBI Taxonomy (https://www.ncbi.nlm.nih.gov/taxonomy; accessed 02/2025-03/2025) and IndexFungorum (https://www.indexfungorum.org/names/names.asp; accessed 02/2025-03/2025). Metadata asscoiated with symbiotic-status were collected from supporting literature and the Consortium of Lichen Herbaria (https://lichenportal.org/portal/index.php). 443 genomes were downloaded from NCBI and JGI, and were filtered for BUSCO single-copy completeness of >= 95% using the ascomycota_odb12 gene models (BUSCO v5.8.2). The remaining 119 genomes were filtered by genomic coverage >= 25x, using reported coverage metrics from NCBI and JGI. The remaining 108 genomes were annotated following the Funannotate2 pipeline (Funannotate2 v24.4.30) (Palmer and Stajich, unpublished). Genomes were cleaned with the Funannotate2 clean module and soft-masked using RepeatMasker (v4.1.7). RepeatMasker was run using the --xsmall flag (soft-mask), the NCBI/RMBLAST search engine (v2.14.1+), and the Fungi Dfam RepeatMasker database (Dfam v3.8; 11/14/2023). Gene prediction programs were trained using the soft-masked genomes, using the Funannotate2 training module. Then, genome structure and gene prediction was analyzed with Funannotate2 predict module. Gene models were annotated with Funannotate2 annotate module.

The outputs of the funannotate2 annotate module were re-analyzed using InterProScan (v__), antiSMASH (v__), and SignalP (v__). The outputs of these analyses were incorporated into the existing annotations using Funannotate 1.8.15 annotate module, which also runs Diamond BLASTp, EggNogMapper (v__) MEROPS, and UniProt. The results from antiSMASH are also cross-referenced with the MiBiG database (v 1.4). The results from the this accessory analysis were fed back into funannotate2 annotate module for final reconcilliation of annotations. 

Orthologs were identified with OrthoFinder (v3.0.1b1) using the annotated protein FASTA files from the Funannotate2 pipeline. OrthoFinder was run using DIAMOND (v2.1.11) as the search engine, MAFFT (v7.525) for the multiple sequence alignment (-MSA flag) and IQTree (v2.4.0) (-T flag) was used to create gene trees for orthogroups. 1030 orthogroups with minimum of 98.1% of species having single-copy genes in any orthogroup were used to infer the species tree. 
